go/worker/client/worker.go

```diff
@@ -58,8 +59,13 @@ func (w *Worker) Start() error {
 
 	// Wait for all runtimes to be initialized.
 	go func() {
-		for _, rt := range w.runtimes {
+		for id, rt := range w.runtimes {
 			<-rt.Initialized()
```

Comment:
Are you sure this is waiting long enough? An observer should only be deemed initialized once the runtimes are fully provisioned.

See the executor (`go/worker/compute/executor/committee/node.go`) as an example.
https://github.com/oasisprotocol/oasis-core/blob/299a63f08e5a511a0dc7f9987cbef51cb28cf543/go/worker/compute/executor/committee/node.go#L954-L984

Reply:
I guess what the executor is doing is overkill for the client, but it should at least wait for the runtime to be started, e.g. after receiving this event:
https://github.com/oasisprotocol/oasis-core/blob/299a63f08e5a511a0dc7f9987cbef51cb28cf543/go/worker/common/committee/node.go#L588-L602

Reply:
You could handle this in the client event hook:
https://github.com/oasisprotocol/oasis-core/blob/299a63f08e5a511a0dc7f9987cbef51cb28cf543/go/worker/client/committee/node.go#L91-L94

Reply:
You are right, `initCh` is closed almost immediately, not waiting the runtime to start. Maybe we need to change this, because the common worker has a different approach.

https://github.com/oasisprotocol/oasis-core/blob/master/go/worker/common/committee/node.go#L728-L729

Reply:
The client worker first waits for the common worker to initialize, so this is already taken into account. But this still does not wait for the runtime to initialize.